File: Problem Set 1/mario.c
// Prompts the user for a height between 1 and 8 and prints two aligned pyramids with a gap between them.

#include <stdio.h>
void printPyramid(int n);

int main(void)
{
    while (1)
    {
        int n;

        printf("How tall should the pyramid be (1, 8 inclusive)? ");
        scanf("%i", &n);

        if (n > 0 && n < 9)
        {
            printPyramid(n);
            break;
        }
    }

    return 0;
}

void printPyramid(int n)
{
    int i;
    int k;
    int x;
    int y = n - 1;

    for (i = 0; i < n; i++)
    {

        for (x = 0; x < y; x++)
        {
            printf(" ");
        }

        for (k = 0; k < i + 1; k++)
        {
            printf("#");
        }

        printf("  ");

        for (k = 0; k < i + 1; k++)
        {
            printf("#");
        }

        printf("\n");
        y--;
    }

    return;
}

File: Problem Set 6 /mario.py
# Prompt the user for a height between 1 and 8 (inclusive), repeatedly asking until valid input is given.
# Print two side-by-side pyramids of that height using '#', aligned properly with spaces and a gap of 2 between them.

def main():
    height = get_height()

    for i in range(1, height + 1):
        spaces = " " * (height - i)
        blocks = "#" * i
        print(spaces + blocks + "  " + blocks)


def get_height():
    while True:
        try:
            height = int(input("Enter height: "))
            if 1 <= height <= 8:
                return height
        except ValueError:
            pass


if __name__ == "__main__":
    main()
